code/datums/status_effects/wound_effects.dm

```diff
@@ -30,6 +30,7 @@
 	status_type = STATUS_EFFECT_REPLACE
 	tick_interval = STATUS_EFFECT_NO_TICK
 	alert_type = /atom/movable/screen/alert/status_effect/limp
+	on_remove_on_mob_delete = TRUE
```

Comment:
So on_remove is called on human qdel. This happens before leg qdel in the chain so the flaky leg harddels should be fixed. Previously it just did not clean references, guaranteeing a harddel